treescript/tests/test_mercurial.py

```diff
@@ -315,23 +423,17 @@ def info(msg, *args):
     assert is_slice_in_list(("push", "-e", expect), called_args[0][0])
 
 
-# log_outgoing {{{1
 @pytest.mark.asyncio
-async def test_log_outgoing(config, task, mocker):
-    called_args = []
+async def test_push_fail(config, task, mocker, tmpdir):
+    """Raise a PushError in run_hg_command and verify we clean up."""
 
-    async def run_command(config, *arguments, repo_path=None, **kwargs):
-        called_args.append([tuple([config]) + arguments, {"repo_path": repo_path}])
+    async def blow_up(*args, **kwargs):
+        raise PushError("x")
 
-    mocker.patch.object(mercurial, "run_hg_command", new=run_command)
-    mocked_source_repo = mocker.patch.object(mercurial, "get_source_repo")
-    mocked_source_repo.return_value = "https://hg.mozilla.org/treescript-test"
-    await mercurial.log_outgoing(config, task, config["work_dir"])
+    async def clean_up(*args):
+        assert args == (config, task, tmpdir)
 
-    assert len(called_args) == 1
-    assert "repo_path" in called_args[0][1]
-    assert is_slice_in_list(("out", "-vp"), called_args[0][0])
-    assert is_slice_in_list(("-r", "."), called_args[0][0])
-    assert is_slice_in_list(
-        ("https://hg.mozilla.org/treescript-test",), called_args[0][0]
-    )
+    mocker.patch.object(mercurial, "run_hg_command", new=blow_up)
+    mocker.patch.object(mercurial, "strip_outgoing", new=clean_up)
+    with pytest.raises(PushError):
+        await mercurial.push(config, task, tmpdir)
```

Comment:
note, this looks like if we fail to actually run clean_up, we're not *actually* going to fail here...

we might need something like `mocked_strip.assert_called_once` after the `with pytest.raises()` block.

Reply:
Done.